src/syncDaemon.ts

```diff
@@ -47,7 +47,8 @@ console.log('Cron daemon is started');
 
 				const progressRepository: ProgressRepository = getConnection().getCustomRepository(ProgressRepository);
 				for (const contract of contracts) {
-					for (const event of events) {
+					for (const eventId of (contract.events || [])) {
+						const event = events.find((e) => e.eventId === eventId);
```

Comment:
Please create separate function getEventsByContractId

Reply:
Should I do additional SQL request instead of js func?